sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -47,6 +48,15 @@ trait Predicate extends Expression {
 
 
 trait PredicateHelper {
+
+  protected def partitionByDeterminism(predicates: Seq[Expression]) = {
+    if (SQLConf.get.outOfOrderPredicateEvaluationEnabled) {
+      predicates.partition(_.deterministic)
+    } else {
+      predicates.span(_.deterministic)
+    }
+  }
+
```

Comment:
Is not this "if" in the other way around here? As Seq#partition keeps the relative orders and Seq#span does not.
After reversing this "if" the "outOfOrderPredicateEvaluationEnabled" default value (true) will be consistent with the old behavior, before your PR (I mean using the Seq#span), but in the unit tests the relevant "if" conditions should be changed too.